model/src/pyrenew/metaclass.py

```diff
@@ -339,6 +340,12 @@ def run(
                 nuts_args=nuts_args,
                 mcmc_args=mcmc_args,
             )
+        if rng_key is None:
+            rand_int_key = jr.PRNGKey(0)
+            rand_int = jr.randint(
```

Comment:
Doesn't this guarantee that any time `rng_key` is `None`, we will generate the same `rand_in` and therefore get the same samples from the model? Is that what is desired?